docs/cookbook.md

```diff
@@ -0,0 +1,412 @@
+
+## Installation :
+
+```bash
+composer --prefer-dist require akeneo-labs/custom-entity-bundle 2.0.*
+```
+
+Follow the bundle documentation : https://github.com/akeneo-labs/CustomEntityBundle/blob/master/README.md#installation
+
+## Create the reference data
+
+To create your entity, you have to create a class that extends AbstractCustomEntity. In this cookbook, we will create an entity called Supplier with only one property `name`.
+
+AbstractCustomEntity already declares 2 mandatory properties : `code` and `id`, so you just have to add your entity specific properties.
+
+```php
+<?php
+
+namespace Acme\Bundle\AppBundle\Entity;
+
+use Pim\Bundle\CustomEntityBundle\Entity\AbstractCustomEntity;
+
+class Supplier extends AbstractCustomEntity
+{
+    /**
+    * @var string
+     */
+    protected $name;
+    
+    public function getName(): string
+    {
+        return $this->name;
+    }
+
+    public function setName($name): void
+    {
+        $this->name = $name;
+    }
+}
+```
+
+Then, create the doctrine configuration for this entity to declare the entity fields and table name where the entity will be persisted.
+
+```yaml
+#Acme\Bundle\SupplierBundle\Resources\config\doctrine\Supplier.orm.yml
+Acme\Bundle\SupplierBundle\Entity\Supplier:
+    type: entity
+    table: refdata_supplier
+    repositoryClass: Pim\Bundle\CustomEntityBundle\Entity\Repository\CustomEntityRepository
+    fields:
+        id:
+            type: integer
+            id: true
+            generator:
+                strategy: AUTO
+        code:
+            type: string
+            length: 25
+            nullable: false
+            unique: true
+        name:
+            type: string
+            length: 255
+            nullable: false
+```
+
+## Update the database
+
+Each new reference data is persisted in a dedicated table. You can see what will be the create table SQL query by executing this command :
+```bash
+bin/console doctrine:schema:update --dump-sql
+```
+
+In our case, this will output something like this : 
+```sql
+CREATE TABLE refdata_supplier (id INT AUTO_INCREMENT NOT NULL, code VARCHAR(25) NOT NULL, name VARCHAR(255) NOT NULL, UNIQUE INDEX UNIQ_EF5B01F877153098 (code), PRIMARY KEY(id));
+```
+
+Now run the command by replacing the `--dump-sql` option by `--force` to create the table in the database :
+```bash
+bin/console doctrine:schema:update --force
+```
+
+You can check that the table is correctly created in your database.
+ 
+## Your reference data as an attribute
+Now declare the new reference data in the application : 
+```yaml
+#app/config/config.yml
+pim_reference_data:
+    supplier:
+        class: Acme\Bundle\SupplierBundle\Entity\Supplier
+        type: simple
+```
+
+To check every thing is OK so far, execute the following commands :
+```bash
+#check if the entity doctrine configuration is OK
+bin/console doctrine:mapping:info
+```
+```bash
+#check if the reference data is correct
+bin/console pim:reference-data:check
+```
```

Comment:
From L88 to 105 could be replaced by this article that already exists: https://docs.akeneo.com/2.0/manipulate_pim_data/catalog_structure/creating_a_reference_data.html#configuring-the-reference-data
It avoids creating support on an external subject.